express/index.cjs

```diff
@@ -347,11 +366,172 @@ app.post('/login/employer', async (req, res) => {
+    }
+
+     // Use checkUser helper function
+     const user = await checkUser(req, email);
+
+     if (!user.exists) {
+      throw({status: 404, success: false, error: 'failed reset attempt: password', reason: 'email not found'});
+     }
+
+    const { usertype, userId } = user;
+
+    // Create a reset token
+    const resetToken = jwt.sign({email: email, id: userId, type: usertype}, process.env.JWT_KEY, { expiresIn: '1h' });
+    const resetLink = `http://localhost:5173/reset-password?token=${resetToken}`;
+
+    await sendEmail(email, 'Password Reset', `Click here to reset your password: ${resetLink}`);
+
+    writer.write(`${setTimestamp(newTime)} | status: 200 | source: /forgot-password | success | Email successfully sent || ${email}@${req.socket.remoteAddress}\n `);
+
+    res.status(200).json({ success: true, message: 'Reset link sent to email', token: resetToken });
+  } catch (err) {
+    console.warn(err);
+    if(!err.reason) {
+      res.status(500).json({ success: false, error: 'Server failure' });
+      writer.write(`${setTimestamp(newTime)} | status: 500 | source: /forget-password | error: ${err} || attempt: ${email}@${req.socket.remoteAddress}\n`);
+    } 
+    else {
+      res.status(!err.status ? 500 : err.status).json({success: false, error: err.reason});
+      writer.write(`${setTimestamp(newTime)} | status: ${!err.status ? 500 : err.status} | source: /forget-password | error: ${err.error} | reason: ${err.reason} | attempt: ${email}@${req.socket.remoteAddress}\n`);
+    }   
+  }
+});
+
+// reset password
+app.put('/reset-password', async (req, res) => {
+  console.log('reset password attempt')
+  const newTime = new Date(Date.now());
+  const { token, newPassword } = req.body;
+  let userId, usertype, email;
+  writer.write(`${setTimestamp(newTime)} || source: /reset-password | info: resetting password || attempt: ${email}@${req.socket.remoteAddress}\n`);
+
+  try {
+    if (!token || !newPassword) {
+      throw({status: 400, success: false, error: 'failed reset attempt: password', reason: 'missing token or new password'});
+    }
+
+    const user = jwt.verify(token, process.env.JWT_KEY);
+    userId = user.id;
+    usertype = user.type;
+    email = user.email;
+
+    const hash = await bcrypt.hash(newPassword, 10);
+
+    let query, params;
+
+    if (usertype === 'seeker') {
+      query = 'UPDATE Seeker SET user_pass = :hash WHERE seeker_id = UNHEX(:userId) AND delete_flag = 0;';
+      params = { hash, userId };
+    } else {
+      query = 'UPDATE Employer SET user_pass = :hash WHERE employer_id = UNHEX(:userId) AND delete_flag = 0;';
+      params = { hash, userId };
+    }
+
+    const result = await req.db.query(query, params);
+
+    writer.write(`${setTimestamp(newTime)} | status: 200 | source: /reset-password | success: Password successfully reset || ${userId}@${req.socket.remoteAddress}\n`);
```

Comment:
Why is userId used here while others use email in the same function?

Reply:
that was a mistake, just changed it to email